persist/persist_test.go

```diff
@@ -26,3 +26,69 @@ func TestIntegrationRandomSuffix(t *testing.T) {
 		file.Close()
 	}
 }
+
+// TestUnitNewSafeFile checks that a new file is created and that its name is
+// different than the finalName of the new safeFile.
+func TestUnitNewSafeFile(t *testing.T) {
+	// Create safe file.
+	sf, err := NewSafeFile("NewSafeFile test file" + RandomSuffix())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(sf.Name())
+	defer sf.Close()
+
+	// Check that the name of the file is not equal to the final name of the
+	// file.
+	if sf.Name() == sf.finalName {
+		t.Errorf("safeFile temporary filename and finalName are equivalent: %s\n", sf.Name())
+	}
+}
+
+// testSafeFileWithPath tests creating and committing safe files with a
+// specified path.
+func testSafeFileWithPath(filename string, t *testing.T) {
```

Comment:
we prefer to avoid passing `testing.T` around to functions because it can obscure the exact nature of the failure (e.g. multiple failing tests will cite the same line number). Instead, have the function return an error and handle the error in the actual testing function.

alternatively, these two tests might be better off as a table-driven test anyway.